internal/backend/bls12-377/groth16/prove.go

```diff
@@ -62,8 +63,10 @@ func Prove(r1cs *cs.R1CS, pk *ProvingKey, fullWitness witness.Witness, opts ...b
 
 	proof := &Proof{}
 
+	solverOpts := opt.SolverOpts[:len(opt.SolverOpts):len(opt.SolverOpts)]
+
 	if r1cs.CommitmentInfo.Is() {
-		opt.HintFunctions[r1cs.CommitmentInfo.HintID] = func(_ *big.Int, in []*big.Int, out []*big.Int) error {
+		solverOpts = append(solverOpts, solver.OverrideHint(r1cs.CommitmentInfo.HintID, func(_ *big.Int, in []*big.Int, out []*big.Int) error {
```

Comment:
This is an interesting way to go. Should we settle on this permanently or is it temporary until we figure out a cleaner way?

Reply:
It started as temporary for the refactor, then I thought "well why not, ..." . So ... don't know :)